clients/src/main/java/org/apache/kafka/clients/consumer/internals/RequestManagers.java

```diff
@@ -71,6 +72,7 @@ public RequestManagers(LogContext logContext,
         this.commitRequestManager = commitRequestManager;
         this.topicMetadataRequestManager = topicMetadataRequestManager;
         this.fetchRequestManager = fetchRequestManager;
+        this.shareFetchRequestManager = null;
```

Comment:
Shall we have the member as Optional then?

Reply:
The inconsistency between Optional and non-Optional fields in this class bothers me a bit. It's shared with the async consumer, so the inconsistency comes from there really. I think when we merge into AK, either everything will be Optional, or nothing will.